src/ui/public/vislib/lib/axis_config.js

```diff
@@ -53,23 +54,23 @@ export default function AxisConfigFactory() {
   };
 
   class AxisConfig {
-    constructor(config) {
-      const typeDefaults = config.type === 'category' ? categoryDefaults : {};
-      this._values = _.defaultsDeep({}, config, typeDefaults, defaults);
+    constructor(chartConfig, axisConfig) {
+      const typeDefaults = axisConfig.type === 'category' ? categoryDefaults : {};
+      this._values = _.defaultsDeep({}, axisConfig, typeDefaults, defaults);
 
       this._values.elSelector = this._values.elSelector.replace('{pos}', this._values.position);
-      this._values.rootEl = this._values.vis.el;
+      this._values.rootEl = chartConfig.get('el');
 
-      this.data = this._values.data;
+      this.data = chartConfig.data;
       if (this._values.type === 'category') {
         this.values = this.data.xValues();
         this.ordered = this.data.get('ordered');
       }
 
       if (this._values.type === 'value') {
-        const isWiggleOrSilluete = this._values.vis._attr.mode === 'wiggle' || this._values.vis._attr.mode === 'silluete';
+        const isWiggleOrSilluete = chartConfig.mode === 'wiggle' || chartConfig.mode === 'silluete';
```

Comment:
what is this wiggle/silluete thing ? is this something you can apply to single series (of the same axis) ? is this something you apply to axis ? or is this something you apply to chart in general ?